frontend/gateway/forwarder/forward.go

```diff
@@ -113,6 +118,59 @@ func (c *bridgeClient) Inputs(ctx context.Context) (map[string]llb.State, error)
 	return inputs, nil
 }
 
+func (c *bridgeClient) wrapSolveError(solveErr error) error {
+	var (
+		ee        *llberrdefs.ExecError
+		fae       *llberrdefs.FileActionError
+		sce       *solver.SlowCacheError
+		inputIDs  []string
+		outputIDs []string
+		subject   errdefs.IsSolve_Subject
+	)
+	if errors.As(solveErr, &ee) {
+		var err error
+		inputIDs, err = c.registerResultIDs(ee.Inputs...)
+		if err != nil {
+			return err
```

Comment:
In the edge case that the `registerResultIDs` returns the `unexpected type for result` error then we will lose the original solve error here.  I wonder if these should be `errors.Wrap(solveErr, err.Error())` instead to make sure the solve error is preserved?  I am not sure if we would ever practically get the `unexpected type for result` error though.

Reply:
> I am not sure if we would ever practically get the unexpected type for result error though.
I think that would be an implementation error in Buildkit?

I'm okay with the `errors.Wrap` approach, but unsure if it'll be useful if incomplete? You'll need a lot of safeguards in the client side for incomplete solve errors.